### PR TITLE
Lower AWS DescribeVolume frequency

### PR DESCRIPTION
**Pull request by jsafrane**
**What this PR does / why we need it**:
Our (OpenShift) runs multiple e2e tests (i.e. multiple Kubernetes clusters) in a single AWS project. Kubernetes under e2e tests exercises AWS API a lot and sometimes it exceeds API quota and test flakes.

`DescribeVolumes` has been found as the most throttled request. This PR add initial delay and lowers the frequency of DescribeVolumes polling when waiting for attach/detach to finish.

After a week of running this code in our CI:
* Average attach time went down from 4.09s to 3.70 (0.39s improvement on average).
  * Calculated as `rate(storage_operation_duration_seconds_sum{volume_plugin="
kubernetes.io/aws-ebs", operation_name="volume_attach"}[1y]) / rate(storage_operation_duration_seconds_count{volume_plugin="kubernetes.io/aws-ebs", operation_
name="volume_attach"}[1y]` ("1y" to cover all captured metrics).
  * It is harder to get 95-percentile from prometheus metrics, I can get it if you're interested and you give me the right query :-).
* Average detach time went up from 7.95s to 8.12s (0.17s slower on average)
* Number of DescribeVolumes calls from both went from 702.07 calls/e2e job to 521.82 calls/e2e job on average (saved 25.67% of calls).
  * Calculated from `cloudprovider_aws_api_request_duration_seconds_count{request="describe_volume"}[1y]`
* I don't have exact numbers of DescribeVolumes `RequestLimitExceeded` errors, because we run jobs both with and without the PR in the same AWS project, but it went down from 2069 RequestLimitExceeded errors/day (0.44% of all DescribeVolumes calls) to 1166/day (0.25% of all DescribeVolumes calls) for whole project.

The numbers look great, however, I can test it at scale only in our CI. The numbers may be affected by AWS zone (`us-east-1{a,b,c}`) or AWS project. It would be great if someone else could confirm the numbers or at least confirm there is no performance regression in attach/detach time.

@wongma7 @leakingtapan, PTAL.
cc @gnufied @bertinatto 

**Does this PR introduce a user-facing change?**:

```release-note
Reduced frequency of DescribeVolumes calls of AWS API when attaching/detaching a volume.
```

/sig storage
/sig cloud-provider
/kind flake

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 1]
```diff
@@ -2114,6 +2116,12 @@ func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment,
 	// So we tolerate a limited number of failures.
 	// But once we see more than 10 errors in a row, we return the error
 	describeErrorCount := 0
+
+	// Attach/detach usually takes time. It does not make sense to start
+	// polling DescribeVolumes before some initial delay to let AWS
+	// process the request.
+	time.Sleep(getInitialAttachDetachDelay(status))
```
Comment (wongma7) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: very very good idea. 👍

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 2]
```diff
@@ -2114,6 +2116,12 @@ func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment,
...
+
 	var attachment *ec2.VolumeAttachment
 
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
```
Comment (wongma7) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: any ideas for metrics that could help us tune this more in the future, if necessary? How did you figure out that 2s is average?

**Comment by wongma7** [comment]
lgtm. The sleep before calling DescribeVolume seems like a no-brainer in hindsight, very good improvement.

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go (hunk 2 above)
Comment (jsafrane) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go]: I set really high frequency of DescribeVolumes polling and ran bunch of tests.

We still have metric `storage_operation_duration_seconds_sum{volume_plugin=" kubernetes.io/aws-ebs", operation_name="volume_attach"}`, but it includes `DescribeInstance` + `AttachVolume` calls + `DescribeVolumes` polling.

**Comment by bertinatto** [comment]
LGTM too. 

Something similar has been done for volume creation here: https://github.com/kubernetes/kubernetes/pull/78601

**Pull request merged**